lib/portus/registry_client.rb

```diff
@@ -51,16 +51,19 @@ def manifest(repository, tag = "latest")
     #   - name: a string containing the name of the repository.
     #   - tags: an array containing the available tags for the repository.
     def catalog
-      res = perform_request("_catalog")
-      if res.code.to_i == 200
-        catalog = JSON.parse(res.body)
-        add_tags(catalog["repositories"])
-      elsif res.code.to_i == 404
-        handle_error res
-      else
-        raise "Something went wrong while fetching the catalog " \
-          "Response: [#{res.code}] - #{res.body}"
+      last = ""
+      res = []
+
+      # We fetch repositories in pages of 100 because of a bug in the registry.
```

Comment:
I'm commenting on your issue now, but your implementation here ignores a critical portion of the API specification for assembling catalog walks.

> Compliant client implementations should always use the Link header value when proceeding through results linearly. The client may construct URLs to skip forward in the catalog.

https://github.com/docker/distribution/blob/master/docs/spec/api.md#pagination